src/java/main/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -109,6 +110,20 @@ public void channelConnected(ChannelHandlerContext ctx,
                     zkServer, NettyServerCnxnFactory.this);
             ctx.setAttachment(cnxn);
 
+            InetAddress addr = ((InetSocketAddress)cnxn.channel.getRemoteAddress()).getAddress();
+            Set<NettyServerCnxn> s = ipMap.get(addr);
```

Comment:
Good catch! Congrats! :smiley:

Now, I am a bit unsure if this:

* this is a real issue and/or worth fixing;
* how to achieve a good compromise without entangling the logic too much.

Wdyt?